### DISCUSSION TITLE
How to use a SlashCommand to play music in that user who invoked it's channel

### DISCUSSION THREAD

**Discussion by mimic-the-mad** [nextcord/nextcord#981] [Q&A]
I'm trying to find a way for users to use a slash command called play

so it's basically 

/play [string]

and it plays the song.

I'm coming from discord, and CTX does not function the same as the Interaction, and I can not find a reference where a user has working slash commands play music for Nextcord. It seems every other post runs into the same issue, and no one answers.

` @nextcord.slash_command(name="play_test", description="Plays a song by URL",
                            guild_ids=[guild_id])
    async def play_test(self, interaction: nextcord.Interaction, search: str):

        vc = ctx.voice_client

        if not vc:
            await ctx.invoke(self.connect_)

        player = self.get_player(ctx)

        source = await YTDLSource.create_source(ctx, search, loop=self.bot.loop, download=False)

        await player.queue.put(source)`

**Comment by bmaxdk** [comment]
Replace `ctx` with `interaction`. Since interaction doesn't have a `voice_client` attribute, you need to get the voice client from the guild instead. Call `await interaction.response.send_message()` to send a response to the slash command.

**Comment by ghost** [comment]
Use interaction not ctx

**Reply by ooliver1** [reply]
This discussion is over 10 months old. The other answer already gave this.
